sdk/storage/azure-storage-file-datalake/azure/storage/file/out/datalake/_data_lake_directory_client.pyi

```diff
@@ -0,0 +1,23 @@
+# Stubs for datalake._data_lake_directory_client (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._data_lake_file_client import DataLakeFileClient
+from ._models import DirectoryProperties
+from ._path_client import PathClient
+from typing import Any, Optional
+
+class DataLakeDirectoryClient(PathClient):
+    def __init__(self, account_url: str, file_system_name: str, directory_name: str, credential: Optional[Any]=..., **kwargs: Any) -> Any: ...
+    def generate_shared_access_signature(self) -> None: ...
+    @classmethod
+    def from_connection_string(cls: Any, conn_str: str, file_system_name: str, directory_name: str, credential: Optional[Any]=..., **kwargs: Any) -> DataLakeDirectoryClient: ...
+    def create_directory(self, content_settings: Optional[ContentSettings]=..., metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> Dict[str, Union[str, datetime]]: ...
+    def delete_directory(self, **kwargs: Any) -> None: ...
+    def get_directory_properties(self, **kwargs: Any) -> DirectoryProperties: ...
+    def rename_directory(self, rename_destination: Any, **kwargs: Any): ...
+    def create_sub_directory(self, sub_directory: Union[DirectoryProperties, str], content_settings: Optional[ContentSettings]=..., metadata: Optional[Dict[str, str]]=..., **kwargs: Any) -> DataLakeDirectoryClient: ...
```

Comment:
`content_settings` should be a `**kwarg`

Reply:
probably just remove it? content_settings on directory will not be applied to files under the directory.

Reply:
We agreed to remove content_settings parameter from create_directory and create_sub_directory